bench/Libraries/Microsoft.Extensions.Telemetry.PerformanceTests/ModernCodeGen.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Telemetry.Logging;
+
+namespace Microsoft.Extensions.Telemetry.Bench;
+
+internal static partial class ModernCodeGen
+{
+    /// <summary>
+    /// Logs "Connection id '{connectionId}' received {type} frame for stream ID {streamId} with length {length} and flags {flags} and {other}" at "Error" level.
+    /// </summary>
+    [global::System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.Gen.Logging", "8.0.0.0")]
+    public static void RefTypes(global::Microsoft.Extensions.Logging.ILogger logger, string connectionId, string type, string streamId, string length, string flags, string other)
```

Comment:
Why no use a `[LogMethod]` here?

Reply:
This code shows new code generation pattern, but the code generator doesn't yet produce this output. So I hand-wrote it.